core/store/src/main/java/mil/nga/giat/geowave/core/store/DataStore.java

```diff
@@ -267,6 +268,29 @@ public boolean deleteEntry(
 
 	/**
 	 * Returns all data in this data store that matches the query parameter
+	 * within the index described by the index passed in. All data types that
+	 * match the query will be returned as an instance of the native data type
+	 * that was originally ingested, but only the subset of attributes specified
+	 * will be included.
+	 * 
+	 * @param index
+	 *            The index information to query against. All data within the
+	 *            index of this index ID will be queried and returned.
+	 * @param attributes
+	 *            The subset of attributes to include in the results
+	 * @param query
+	 *            The description of the query to be performed
+	 * @return An iterator on all results that match the query. The iterator
+	 *         implements Closeable and it is best practice to close the
+	 *         iterator after it is no longer needed.
+	 */
+	public <T> CloseableIterator<T> query(
```

Comment:
What are the thoughts on just adding a "QueryOptions" object - that way we don't break the API as new bits get added.  
If that answers the mail, then I would say move fieldIds to that (since  it's a pretty minor change).

If we need to do a bigger re-factoring, then i'm fine deferring it to a new ticket.